From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Mon, 24 Aug 2020 19:40:00 +0200
Subject: [PATCH 1/1] Conditional exclude the libpng reader when building with
 libspng

Similar to ce63fc1, but allowing libpng to be configured with
-DPNG_NO_READ (i.e reading functionally disabled).

Upstream-Status: Pending

diff --git a/libvips/foreign/vipspng.c b/libvips/foreign/vipspng.c
index 1111111..2222222 100644
--- a/libvips/foreign/vipspng.c
+++ b/libvips/foreign/vipspng.c
@@ -161,6 +161,8 @@ user_warning_function( png_structp png_ptr, png_const_charp warning_msg )
 	g_warning( "%s", warning_msg );
 }
 
+#ifndef HAVE_SPNG
+
 #define INPUT_BUFFER_SIZE (4096)
 
 /* What we track during a PNG read.
@@ -763,6 +765,8 @@ vips__png_isinterlaced_source( VipsSource *source )
 	return( interlace_type != PNG_INTERLACE_NONE );
 }
 
+#endif /*!defined(HAVE_SPNG)*/
+
 const char *vips__png_suffs[] = { ".png", NULL };
 
 /* What we track during a PNG write.
